src/api/app/models/project/key_info.rb

```diff
@@ -0,0 +1,34 @@
+class Project
+  class KeyInfo
+    include ActiveModel::Model
+
+    attr_accessor :pubkey, :algorithm, :ssl_certificate
+
+    CACHE_EXPIRY_TIME = 5.minutes
+
+    def self.find_by_project_name(project_name)
+      cache_key = "key_info_project_#{project_name}"
```

Comment:
In case of the key probably not a big problem but passing in the project and using 

``
project.cache_key
``

would be more resilient I guess